docs/container-set-template.md

```diff
@@ -0,0 +1,66 @@
+# Container Set Template
+
+![alpha](assets/alpha.svg)
+
+> v3.1 and after
+
+A container set templates is similar to a normal container or script template, but allows you to specify multiple
+containers to run within a single pod.
+
+Because you have multiple containers within a pod, they will be scheduled on the same host. You can use cheap and fast
+empty-dir volumes instead of persistent volume claims to share data between steps.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: container-group-template-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      volumes:
+        - name: workpsace
+          emptyDir: { }
+      containerSet:
+        volumeMounts:
+          - mountPath: /workspace
+            name: workspace
+        graph:
+          containers:
+            - name: a
+              image: argoproj/argosay:v2
+            - name: b
+              image: argoproj/argosay:v2
+            - name: main
+              image: argoproj/argosay:v2
+              dependencies:
+                - a
+                - b
+      outputs:
+        parameters:
+          - name: message
+            valueFrom:
+              path: /workpsace/message
+```
+
+There are a couple of caveats:
+
+1. You must use the [Emissary Executor](workflow-executors.md#emissary-emissary).
+2. Or all containers must run in parallel - i.e. it is a graph with no dependencies.
```

Comment:
Since this is dependent on emissary. I think we should automatically launch this pod using emissary, with no configuration of the controller. The only caveat we would need to document is is that they need to specify command and args, or configure that in the configmap.

Reply:
I actually intentionally avoid this. You can run graphs with other executors, as long as they are not "sequenced". I think that users might like that feature.